### ISSUE TITLE
ktlint violation location in detekt-formatting is broader

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#6105]
## Expected Behavior
Highlight the exact location ktlint reports (│):
```kotlin
private fun getErrorMessage(value: String): String =
        @Suppress("DEPRECATION")
        customMessage.takeUnless { it.isEmpty() }│ ?: String.format(DEFAULT_ERROR_MESSAGE_WITH_NO_REASON, value)
```

## Current Behavior
![image](https://github.com/detekt/detekt/assets/2906988/7836ccb6-83b2-4ff4-aef9-0ba442dd183b)

## Context
While working on a PR we found a ktlint violation in the code that was confusing (see above in current screenshot). This was reported at https://github.com/pinterest/ktlint/issues/2028, and investigated and explained there. The reason it was confusing was because the highlighted code (yellow in above screenshot) didn't match up with the error message "Expected newline". The annotation looked correctly wrapped, but the same rule also checks if the annotated element is nicely separated, hence this emission.

In the explanation there was a line that hit me "This LintError contains the offset (line and position)", and a short investigation lead me to
https://github.com/detekt/detekt/blob/948276a8db4578b962bf24c63eee3b8548bac627/detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/FormattingRule.kt#L97

showing that the offset reported by the ktlint `emit` function is ignored (`_`). This means that **detekt-wrapped ktlint rules will always report less precise location** than if they were directly executed with another wrapper.

This is just one example, there could be many others where the detekt-wrapped reporting is confusing. I understand this is quite a fundamental issue, and might not fit detekt's model, but wanted to record the problem anyway.

**Comment by cortinico** [comment]
> This is just one example, there could be many others where the detekt-wrapped reporting is confusing. I understand this is quite a fundamental issue, and might not fit detekt's model, but wanted to record the problem anyway.

Thanks for reporting this. I was aware of this misalignment when I worked on KtLint wrapper recently (previously we were reporting a finding on the whole file, now it's on the AST Node level, which is already an improvement).

I remember trying to use the offset information provided by KtLint but the result wasn't good enough (I was getting reports at wrong locations). We definitely need to look again into this.

**Comment by BraisGabin** [comment]
Related with #6390, right? Should that pr close this one?

**Comment by cortinico** [comment]
> Related with #6390, right? Should that pr close this one?

I think this was originally fixed by my PR #5876, but @arturbosch's one #6390 will fix a follow-up bug from it.

**Comment by arturbosch** [comment]
> > Related with #6390, right? Should that pr close this one?
> 
> I think this was originally fixed by my PR #5876, but @arturbosch's one #6390 will fix a follow-up bug from it.

Not necessary all cases are fixed. Imagine KtLint reports offsets for nodes but on a class node. The recent detekt change would use the class node as reporting location e.g. all kdoc + annotations.
So yes, #6390 should fix all these cases as we again report the original KtLint offsets.
